pkg/controllers/management/drivers/kontainerdriver/kontainerdriver.go

```diff
@@ -0,0 +1,409 @@
+		dynamicSchema.Spec.ResourceFields = resourceField
+		dynamicSchema.Spec.Embed = true
+		dynamicSchema.Spec.EmbedType = "cluster"
+		_, err := l.dynamicSchemas.Create(dynamicSchema)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	shouldUpdate := false
+
+	if clusterSchema.Spec.ResourceFields == nil {
+		clusterSchema.Spec.ResourceFields = map[string]v3.Field{}
+	}
+	if _, ok := clusterSchema.Spec.ResourceFields[fieldName]; !ok {
+		// if embedded we add the type to schema
+		clusterSchema.Spec.ResourceFields[fieldName] = v3.Field{
+			Create:   true,
+			Nullable: true,
+			Update:   true,
+			Type:     embeddedType,
+		}
+		shouldUpdate = true
+	}
+
+	if shouldUpdate {
+		_, err = l.dynamicSchemas.Update(clusterSchema)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func toResourceField(name string, flag *types.Flag) (string, v3.Field, error) {
+	field := v3.Field{
+		Create: true,
+		Update: true,
+		Type:   "string",
+	}
+
+	name, err := toLowerCamelCase(name)
+	if err != nil {
+		return name, field, err
+	}
+
+	field.Description = flag.Usage
+
+	if flag.Type == types.StringType {
+		field.Default.StringValue = flag.Value
+
+		if flag.Password {
+			field.Type = "password"
+		} else {
+			field.Type = "string"
+		}
+
+		if flag.Default != nil {
+			field.Default.StringValue = flag.Default.DefaultString
+		}
+	} else if flag.Type == types.IntType {
+		field.Type = "int"
+
+		if flag.Default != nil {
+			field.Default.StringValue = strconv.Itoa(int(flag.Default.DefaultInt))
+		}
+	} else if flag.Type == types.BoolType || flag.Type == types.BoolPointerType {
+		field.Type = "boolean"
+
+		if flag.Default != nil {
+			field.Default.BoolValue = flag.Default.DefaultBool
+		}
+	} else if flag.Type == types.StringSliceType {
+		field.Type = "array[string]"
+
+		if flag.Default != nil {
+			field.Default.StringSliceValue = flag.Default.DefaultStringSlice.Value
+		}
+	} else {
+		return name, field, fmt.Errorf("unknown type of flag %v: %v", flag, reflect.TypeOf(flag))
+	}
+
+	return name, field, nil
+}
+
+func toLowerCamelCase(nodeFlagName string) (string, error) {
+	flagNameParts := strings.Split(nodeFlagName, "-")
+	flagName := flagNameParts[0]
+	for _, flagNamePart := range flagNameParts[1:] {
+		flagName = flagName + strings.ToUpper(flagNamePart[:1]) + flagNamePart[1:]
+	}
+	return flagName, nil
+}
+
+func binaryPath(obj *v3.KontainerDriver) string {
+	// TODO this needs to build the binary path based on the unpacked binary name
+	urlPath := strings.Split(obj.Spec.URL, "/")
+	return DriverDir + urlPath[len(urlPath)-1]
+}
+
+func (l *Lifecycle) Updated(obj *v3.KontainerDriver) (runtime.Object, error) {
+	logrus.Infof("update kontainerdriver %v", obj.Name)
+	if obj.Spec.BuiltIn {
+		return obj, nil
+	}
+
+	if hasStaticSchema(obj) {
+		return obj, nil
+	}
+
+	// redownload file if url changed or not downloaded
+	var err error
+	if obj.Spec.URL != obj.Status.ActualURL || v3.KontainerDriverConditionDownloaded.IsFalse(obj) {
```

Comment:
add checkt to see if driver exists so that if we are in an ha scenario and we've migrated to a different server, we can redownload

Reply:
Done.